### PR TITLE
Token airdrop

### PR DESCRIPTION
**Pull request by ponyjackal**

### CODE REVIEW

File: src/heuristics/tokenAirdrop.ts [hunk 1]
```diff
@@ -21,28 +30,39 @@ export function detectTokenAirdrop(transaction: Transaction): boolean {
...
+    if (address === sendAddresses[0]) {
+      continue;
     }
-    airdropTracker[transferKey].amount++;
-    airdropTracker[transferKey].assetTransfers.add(assetTransfer);
 
-    if (airdropTracker[transferKey].amount > AIRDROP_THRESHOLD) {
-      return true;
+    const sent = transaction.netAssetTransfers[address].sent;
+    const received = transaction.netAssetTransfers[address].received;
```
Comment (pcowgill) [inline src/heuristics/tokenAirdrop.ts:47]:
```suggestion
    const sent = transaction.netAssetTransfers[address]?.sent;
    const received = transaction.netAssetTransfers[address]?.received;
```

File: src/heuristics/tokenAirdrop.ts [hunk 2]
```diff
@@ -21,28 +30,39 @@ export function detectTokenAirdrop(transaction: Transaction): boolean {
...
+    if (sent.length > 0 || received.length === 0) {
```
Comment (pcowgill) [inline src/heuristics/tokenAirdrop.ts:48]:
```suggestion
    if (sent && sent.length > 0 || received && received.length === 0) {
```

Comment (pcowgill) [reply src/heuristics/tokenAirdrop.ts:48]: Otherwise it might throw an error when checking the length

**Review by pcowgill — CHANGES_REQUESTED** [review]
@ponyjackal I added some suggestions we should include before merging this.

**Review by jordanmessina — APPROVED** [review]
Paul's suggestions are great.

File: src/heuristics/tokenAirdrop.ts (hunk 1 above)
Comment (ponyjackal) [reply src/heuristics/tokenAirdrop.ts:47]: image

Since I am iterating the key values in the object, the above optional chainings are pointless, I think

**Review by pcowgill — APPROVED** [review]

**Comment by pcowgill** [comment]
The `npm run run:contextualizers` script had a passing result
Screenshot 2023-12-04 at 8 43 51 PM

**Pull request merged**